### PR TITLE
Fix LLVM_jll's deps and compat info

### LINKED ISSUE #2: Dependency warnings
Not sure whether this is the right place to report, but I keep getting this warning when I'm trying to use LLVM_jll
```
pkg> add LLVM_jll@6.0.1
julia> using LLVM_jll
[ Info: Precompiling LLVM_jll [86de99a1-58d6-5da7-8064-bd56ce2e322c]
┌ Warning: Package LLVM_jll does not have Pkg in its dependencies:
│ - If you have LLVM_jll checked out for development and have
│   added Pkg as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with LLVM_jll
└ Loading Pkg into LLVM_jll from project dependency, future warnings for LLVM_jll are suppressed.
```

### PR DESCRIPTION
**Pull request by Gnimuc**
Continuation of https://github.com/JuliaRegistries/General/pull/8398, fix https://github.com/JuliaBinaryWrappers/LLVM_jll.jl/issues/2

cc @vchuravy @giordano @DilumAluthge

### CODE REVIEW

File: L/LLVM_jll/Compat.toml [hunk 1]
```diff
@@ -1,2 +1,8 @@
+[6]
+julia = "1.3.0-1.4.0"
+
+[8]
+julia = "1.4.0-1.5.0"
+
 [9]
-julia = "1"
+julia = "1.5.0-1"
```
Comment (giordano) [inline L/LLVM_jll/Compat.toml]: This was correct. It must be the same also for the other versions

File: L/LLVM_jll/Deps.toml [hunk 2]
```diff
@@ -1,3 +1,3 @@
-[9]
+[6]
```
Comment (giordano) [inline L/LLVM_jll/Deps.toml]: I think we should have a section for each major version, all the sections will be the same

**Review by giordano — CHANGES_REQUESTED** [review]

File: L/LLVM_jll/Compat.toml (hunk 1 above)
Comment (Gnimuc) [reply L/LLVM_jll/Compat.toml]: So there're no compatibility issues between LLVM_jll and Julia? I think LLVM_jll-v6.0.1 cannot be dlopen-ed in Julia-v1.4. This is based on [some old experiments](https://github.com/staticfloat/LLVMBuilder/issues/34), maybe not true anymore.

File: L/LLVM_jll/Deps.toml (hunk 2 above)
Comment (Gnimuc) [reply L/LLVM_jll/Deps.toml]: It looks like the dependencies in the Deps.toml file are increasingly added: https://github.com/JuliaRegistries/General/blob/master/D/DiffEqBase/Deps.toml

File: L/LLVM_jll/Compat.toml (hunk 1 above)
Comment (giordano) [reply L/LLVM_jll/Compat.toml]: I'm not sure what you're talking about, that link is from 2018.

All JLL packages can actually run in v1.3+, but we allow them to be installed in any v1 release

Comment (Gnimuc) [reply L/LLVM_jll/Compat.toml]: If a JLL package cannot be `using` in some Julia versions, should we allow users to install it? Currently, LLVM_jll-v9.0.1 can be installed in Julia-v1.3, but it fails to be used:
```julia
julia> using LLVM_jll
[ Info: Precompiling LLVM_jll [86de99a1-58d6-5da7-8064-bd56ce2e322c]
ERROR: InitError: could not load library "/Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib"
dlopen(/Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib, 1): Symbol not found: __ZN4llvm17raw_pwrite_stream6anchorEv
  Referenced from: /Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib
  Expected in: /Applications/Julia-1.3.app/Contents/Resources/julia/lib/julia/libLLVM.dylib
 in /Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib
Stacktrace:
 [1] #dlopen#3(::Bool, ::typeof(Libdl.dlopen), ::String, ::UInt32) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109
 [2] dlopen at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109 [inlined] (repeats 2 times)
 [3] __init__() at /Users/gnimuc/.julia/packages/LLVM_jll/KqgSs/src/wrappers/x86_64-apple-darwin14-cxx11.jl:215
 [4] _include_from_serialized(::String, ::Array{Any,1}) at ./loading.jl:692
 [5] _require_from_serialized(::String) at ./loading.jl:743
 [6] _require(::Base.PkgId) at ./loading.jl:1034
 [7] require(::Base.PkgId) at ./loading.jl:922
 [8] require(::Module, ::Symbol) at ./loading.jl:917
during initialization of module LLVM_jll
```

Comment (giordano) [reply L/LLVM_jll/Compat.toml]: > If a JLL package cannot be using in some Julia versions, should we allow users to install it?

Yes

Comment (Gnimuc) [reply L/LLVM_jll/Compat.toml]: Oh, never mind. I now know what you mean.

**Comment by vchuravy** [comment]
Ah, I missed those when doing the manual register! Thanks for tracking this down

**Comment by vchuravy** [comment]
Ah, I missed those when doing the manual register! Thanks for tracking this down

**Review by DilumAluthge — APPROVED** [review]
It looks correct to me.

Before merging, can you make sure that doing e.g. `] add LLVM_jll@6`, `] add LLVM_jll@8`, `] add LLVM_jll@9` works fine when using this fork/branch of the registry? Just to make sure that this PR doesn’t break anything.

**Comment by Gnimuc** [comment]
@DilumAluthge I get a warning but I guess it should not be harmful:

### 6
```
(v1.3) pkg> activate mytest
Activating new environment at `~/mytest/Project.toml`

(mytest) pkg> add LLVM_jll@6
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
┌ Warning: Some registries failed to update:
│     — /Users/gnimuc/.julia/registries/General — branch origin/pr/9542 not found
└ @ Pkg.Types /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Types.jl:1199
 Resolving package versions...
  Updating `~/mytest/Project.toml`
  [86de99a1] + LLVM_jll v6.0.1+0
  Updating `~/mytest/Manifest.toml`
  [86de99a1] + LLVM_jll v6.0.1+0
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 

julia> using LLVM_jll

julia> 
```
### 9
```julia
(v1.3) pkg> activate mytest
Activating environment at `~/mytest/Project.toml`

(mytest) pkg> add LLVM_jll@9
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
┌ Warning: Some registries failed to update:
│     — /Users/gnimuc/.julia/registries/General — branch origin/pr/9542 not found
└ @ Pkg.Types /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Types.jl:1199
 Resolving package versions...
  Updating `~/mytest/Project.toml`
  [86de99a1] ↑ LLVM_jll v6.0.1+0 ⇒ v9.0.1+1
  Updating `~/mytest/Manifest.toml`
  [86de99a1] ↑ LLVM_jll v6.0.1+0 ⇒ v9.0.1+1

julia> using LLVM_jll
ERROR: InitError: could not load library "/Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib"
dlopen(/Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib, 1): Symbol not found: __ZN4llvm17raw_pwrite_stream6anchorEv
  Referenced from: /Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib
  Expected in: /Applications/Julia-1.3.app/Contents/Resources/julia/lib/julia/libLLVM.dylib
 in /Users/gnimuc/.julia/artifacts/6520eab5d5ae46f4cccb6010d2390c2ebd130561/lib/libclang.dylib
Stacktrace:
 [1] #dlopen#3(::Bool, ::typeof(Libdl.dlopen), ::String, ::UInt32) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109
 [2] dlopen at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109 [inlined] (repeats 2 times)
 [3] __init__() at /Users/gnimuc/.julia/packages/LLVM_jll/KqgSs/src/wrappers/x86_64-apple-darwin14-cxx11.jl:215
 [4] _include_from_serialized(::String, ::Array{Any,1}) at ./loading.jl:692
 [5] _require_search_from_serialized(::Base.PkgId, ::String) at ./loading.jl:776
 [6] _require(::Base.PkgId) at ./loading.jl:1001
 [7] require(::Base.PkgId) at ./loading.jl:922
 [8] require(::Module, ::Symbol) at ./loading.jl:917
during initialization of module LLVM_jll

julia> 
```

**Review by giordano — APPROVED** [review]

**Pull request merged**